nncf/quantization/advanced_parameters.py

```diff
@@ -193,12 +193,16 @@ class AdvancedAccuracyRestorerParameters:
     :param num_ranking_processes: The number of parallel processes that are used to rank
         quantization operations.
     :type num_ranking_processes: Optional[int]
+    :param intermediate_model_dir: Path to the folder where the model, which was fully
+        quantized with initial parameters, should be saved.
+    :type intermediate_model_dir: Optional[str]
     """
 
     max_num_iterations: int = sys.maxsize
     tune_hyperparams: bool = False
     ranking_subset_size: Optional[int] = None
     num_ranking_processes: Optional[int] = None
+    intermediate_model_dir: Optional[str] = None
```

Comment:
Do we really need this variable? Why not save next to the `optimized` folder?

Reply:
The `intermediate_model_dir` folder is created near the optimized folder on CI. Please refer to build `10569`.

This parameter is required because we need to save the model inside the algorithm's code, and we don't know where to save it. Another solution is to hide this and use `os.environ` to pass the path.